winfsp_wrs/src/callback.rs

```diff
@@ -12,11 +14,75 @@ use crate::{
         SECURITY_INFORMATION, SIZE_T, UINT32, UINT64, ULONG,
     },
     CleanupFlags, CreateFileInfo, CreateOptions, DirInfo, FileAccessRights, FileAttributes,
-    FileInfo, PSecurityDescriptor, SecurityDescriptor, VolumeInfo,
+    FileContextMode, FileInfo, PSecurityDescriptor, SecurityDescriptor, VolumeInfo,
 };
 
+/// Implement only if necessary at your own risk
+pub trait FileContextKind {
+    fn mode() -> FileContextMode;
+    /// # Safety
+    ///
+    /// Write the data into winfsp's `PVOID *PFileContext`
+    /// This is called in winfsp-wrs after calling FileSystemContext::open
+    unsafe fn write(self, out: *mut PVOID);
+    /// # Safety
+    ///
+    /// Retrieve the data from winfsp's `FileContext`
+    /// This is called in winfsp-wrs before calling FileSystemContext::read/write etc.
+    unsafe fn access(raw: PVOID) -> Self;
+    /// # Safety
+    ///
+    /// Retrieve the data from winfsp's `FileContext`
+    /// This is called in winfsp-wrs before calling FileSystemContext::close
+    unsafe fn access_for_close(raw: PVOID) -> Self;
+}
+
+impl<T> FileContextKind for Arc<T> {
+    fn mode() -> FileContextMode {
+        FileContextMode::Descriptor
```

Comment:
`FileContextMode::Full` (i.e. `UmFileContextIsFullContext=1`) is missing in `FileContextMode`

You should add it to `FileContextMode`: this way it's still possible to use this mode (but it's up to the user to implement `impl<T> FileContextKind for MyFullContextStuff`, which seems okay as we talked earlier)

Reply:
Sure 👍